cloud-chatops/README.md

```diff
@@ -1,27 +1,116 @@
+Events are defined in the `schedule_jobs` function:<br>
+- `run_global_reminder()`: Sends a message to the pull request channel with every open pull request across the repositories in a thread.
+- `run_personal_reminder()`: Sends a message to each user directly with a thread of their open pull requests.
+
 ### Deployment
-The entry point for is in [main.py](src/main.py) which will run the application.<br>
-The required files (below) need to be in the [cloud_chatops]() directory.<br>
-### Functionality
-As of current, the application gets all open pull requests from any Cloud owned repository and will send a message to our pull-request channel about each pull request notifying the author.<br>
-The app runs on an asynchronous loop scheduling each reminder to be sent out on days of our catch-ups (Monday, Wednesday and Friday).<br>
-On Mondays the application will mention users with **@** as to notify them directly. However, on the other 2 days authors will not be mentioned as to not spam people.<br>
+
+Secrets should always be stored in `$HOME/cloud_chatops_secrets` on the host system. Find out more about secrets [here](#Secrets)<br>
+The application can be run from a Docker image or source code. (Assuming running from project root)<br>
+
+#### Docker image:
+You can build the image locally or pull from [STFC Harbor](https://harbor.stfc.ac.uk).<br>
+Note: If you are pulling the image you will need to specify a version tag. 
+The latest version can be found in [version.txt](version.txt)<br>
+- ```shell
+  # Local build and run
+  docker build -t cloud_chatops cloud-chatops
+  docker run -v $HOME/cloud_chatops_secrets/:/usr/src/app/cloud_chatops_secrets/ cloud_chatops -d
+  ```
+- ```shell
+  # Pull from harbor and run
+  docker run -v $HOME/cloud_chatops_secrets/:/usr/src/app/cloud_chatops_secrets/ harbor.stfc.ac.uk/stfc-cloud/cloud-chatops:<version> -d
+  ```
+
+#### Running from source:
+You can run the code from [main.py](src/main.py).<br>
+It's always recommended to create a [virtual environment](https://docs.python.org/3/library/venv.html) 
+for the application to run before installing dependencies.
+- ```shell
+  source my_venv/bin/activate
+  pip3 install -r requirements.txt
+  python3 cloud-chatops/src/main.py prod
+  ```
+
+
+
+
+
+
 ### Requirements:
-The following files need to be present in the working directory of the application.<br>
-- **repos.csv**: A list of repositories owned by `stfc` (e.g. `repo1,repo2,repo3`).<br>
-- **user_map.json**: A dictionary of GitHub usernames to Slack Member IDs (e.g. `{"khalford":"ABC123"}`).<br>
-- **secrets.json**: A dictionary of token names to token values (e.g. `{"SLACK_APP_TOKEN":"123ABC"}`). <br>
-- **maintainer.txt**: A text file containing the maintainer users Slack Member ID.
 
-For `secrets.json` the following token structure can be used:<br>
+Two files required for the deployment of this application: `config.yml` and `secrets.json`.<br>
```

Comment:
The config.yml file points to the template file, I'm not sure if that is by design or you want to link the actual config.yml file.

Reply:
That is correct. We can't provide a config with values here as those values change from deployment to deployment. E.g. a different team would need to look at different repos etc. So there is a template for people to copy.

Reply:
Gotcha, that makes sense, then everything else is all good to go.